homeassistant/components/zha/core/channels/hvac.py

```diff
@@ -35,41 +67,390 @@ async def async_set_speed(self, value) -> None:
+    def min_cool_setpoint_limit(self) -> int:
+        """Minimum cooling setpoint."""
+        if self._min_cool_setpoint_limit is None:
+            return self.abs_min_cool_setpoint_limit
+        return self._min_cool_setpoint_limit
+
+    @property
+    def max_heat_setpoint_limit(self) -> int:
+        """Maximum heating setpoint."""
+        if self._max_heat_setpoint_limit is None:
+            return self.abs_max_heat_setpoint_limit
+        return self._max_heat_setpoint_limit
+
+    @property
+    def min_heat_setpoint_limit(self) -> int:
+        """Minimum heating setpoint."""
+        if self._min_heat_setpoint_limit is None:
+            return self.abs_min_heat_setpoint_limit
+        return self._min_heat_setpoint_limit
+
+    @property
+    def local_temp(self) -> Optional[int]:
+        """Thermostat temperature."""
+        return self._local_temp
+
+    @property
+    def occupancy(self) -> Optional[int]:
+        """Is occupancy detected."""
+        return self._occupancy
+
+    @property
+    def occupied_cooling_setpoint(self) -> Optional[int]:
+        """Temperature when room is occupied."""
+        return self._occupied_cooling_setpoint
+
+    @property
+    def occupied_heating_setpoint(self) -> Optional[int]:
+        """Temperature when room is occupied."""
+        return self._occupied_heating_setpoint
+
+    @property
+    def pi_cooling_demand(self) -> int:
+        """Cooling demand."""
+        return self._pi_cooling_demand
+
+    @property
+    def pi_heating_demand(self) -> int:
+        """Heating demand."""
+        return self._pi_heating_demand
+
+    @property
+    def running_mode(self) -> Optional[int]:
+        """Thermostat running mode."""
+        return self._running_mode
+
+    @property
+    def running_state(self) -> Optional[int]:
+        """Thermostat running state, state of heat, cool, fan relays."""
+        return self._running_state
+
+    @property
+    def system_mode(self) -> Optional[int]:
+        """System mode."""
+        return self._system_mode
+
+    @property
+    def unoccupied_cooling_setpoint(self) -> Optional[int]:
+        """Temperature when room is not occupied."""
+        return self._unoccupied_cooling_setpoint
+
+    @property
+    def unoccupied_heating_setpoint(self) -> Optional[int]:
+        """Temperature when room is not occupied."""
+        return self._unoccupied_heating_setpoint
+
+    @callback
+    def attribute_updated(self, attrid, value):
+        """Handle attribute update cluster."""
+        attr_name = self.cluster.attributes.get(attrid, [attrid])[0]
+        self.debug(
+            "Attribute report '%s'[%s] = %s", self.cluster.name, attr_name, value
+        )
+        setattr(self, f"_{attr_name}", value)
+        self.async_send_signal(
+            f"{self.unique_id}_{SIGNAL_ATTR_UPDATED}",
+            AttributeUpdateRecord(attrid, attr_name, value),
+        )
+
+    async def _chunk_attr_read(self, attrs, cached=False):
```

Comment:
Should we move this to helpers? Or do we not think it will be needed anywhere else?

Reply:
yeah, I was thinking about it. Essentially ATM using climate as a test platform for new approaches, but this should be moved into base Channel class, as it looks like more and more entities need to have a bulk of attributes values initialized/retrieved for entity start up. Even more so, once we'll have multiple entities per cluster for sensor entities.